### PR TITLE
actions: drop Qt5 workflows and rename others

### PR DESCRIPTION
**Pull request by shenlebantongying**
I don't think there are actual Qt5 usages https://github.com/xiaoyifang/goldendict-ng/pull/1595#issuecomment-2177807507

No for Windows/macOS as we pretty much the only distributor.

No for Linux. Debian's oldoldstable should be conservative enough, and it will end LTS support in 2 weeks. Ubuntu 22.04 has Qt6.2. For someone stick with old version of things, they are likely still uses the original GD anyway.

### CODE REVIEW

File: .github/workflows/macos-homebrew-PR-check.yml
Comment (xiaoyifang) [inline .github/workflows/macos-homebrew-PR-check.yml:1]: why remove this file?

Comment (shenlebantongying) [reply .github/workflows/macos-homebrew-PR-check.yml:1]: I already switched macOS release to use CMake.

Testing macOS + qmake build has no value anymore.

CMake build don't have the icu linking problem, also have advantage of assembling the .dmg on local machine.

Comment (xiaoyifang) [reply .github/workflows/macos-homebrew-PR-check.yml:1]: Does the cmake  workflow has check whether the code can be compiled normally on linux and macos?

File: .github/workflows/PR-check-windows.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
     strategy:
       matrix:
         os: [windows-2019]
```
Comment (xiaoyifang) [inline .github/workflows/PR-check-windows.yml:30]: no linux ,macos ?

Comment (shenlebantongying) [reply .github/workflows/PR-check-windows.yml:30]: This is qmake+windows PR check. Will delete this when vcpkg+windows is finished.

File: .github/workflows/macos-homebrew-PR-check.yml
Comment (shenlebantongying) [reply .github/workflows/macos-homebrew-PR-check.yml:1]: Yes, It has linux & macOS build jobs.

https://github.com/xiaoyifang/goldendict-ng/blob/staged/.github/workflows/cmake%20build%20check.yml

https://github.com/xiaoyifang/goldendict-ng/blob/23c4b0d25507a33e40ac87b9b791d4d601a050a2/.github/workflows/PR-check-cmake.yml

Comment (shenlebantongying) [reply .github/workflows/macos-homebrew-PR-check.yml:1]: image

**Comment by xiaoyifang** [comment]
fix the build error ,merge by  yourself when you finished.

other parts look good to me.

**Pull request merged**